data/transactions/logic/TEAL_opcodes.md

```diff
@@ -1560,3 +1560,45 @@ For boxes that exceed 4,096 bytes, consider `box_create`, `box_extract`, and `bo
 | 0 | BlkSeed | []byte |  |
 | 1 | BlkTimestamp | uint64 |  |
 
+
+## ec_add v
+
+- Opcode: 0xe0 {uint8 curve}
+- Stack: ..., A: []byte, B: []byte &rarr; ..., []byte
+- for (curve points A and B) return the curve point A + B
+- **Cost**:  BN254=0 BN254_G1=10 BN254_G2=10 BLS12_381=0 BLS12_381_G1=20 BLS12_381_G2=20
+- Availability: v9
+
+`EC` Curves:
+
+| Index | Name | Notes |
+| - | ------ | --------- |
+| 0 | BN254 | BN254 curve, sometimes called BN256 |
```

Comment:
0 & 3 are not supported for `ec_add`

Reply:
Right, and `ec_pairing_check` only supports 0 and 3.  I suppose I should change things up so that the ops that there is a table of the G1/G2s, and a separate table for `ec_pairing_check`, which will only have 0 and 3 rows from this table.

There's no conceivable operation that would want to take either `BN254` OR `BN254_G1`, right?

Reply:
I've changed this so that only allowable curves are actual G1 and G2 curves of each kind.  No opcode accepts a field like 'BN256` (without the G1 or G2).  The only opcode that _did_ accept such an argument was `ec_pairing_check`. Now, that opcode accepts either the G1 _or_ the G2 form, and expects the named curve's points to be first on the stack, and then the "other" curve's points to be last (top) on stack.